### PR TITLE
Baseline1

### PR DESCRIPTION
**Pull request by YummyMirror**
Merge Request for baseline1 branch

### CODE REVIEW

**Comment by YU-alt-acc** [comment]
Missing User model class

**Comment by YummyMirror** [comment]
User class is presented, but I added remember field to it

File: src/test/java/com/practice/git/page/Navigator.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * @author anatoli.anukevich on 18:59 26-Mar-21
+ */
+public class Navigator extends BasePageObject {
+    public Navigator(WebDriver driver) {
+        super(driver);
+    }
+
+    public void mainPage() {
+        super.open("https://opensource-demo.orangehrmlive.com");
```
Comment (YU-alt-acc) [inline src/test/java/com/practice/git/page/Navigator.java:14]: Why is URL hardcoded?

File: src/test/java/com/practice/git/page/LoginPageObject.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ */
+public class LoginPageObject extends BasePageObject {
+    public LoginPageObject(WebDriver driver) {
+        super(driver);
+    }
+
+    public void loginAs(User user) {
+        super.input(By.id("txtUsername"), user.getUsername());
+        super.input(By.id("txtPassword"), user.getPassword());
+        super.click(By.id("btnLogin"));
```
Comment (YU-alt-acc) [inline src/test/java/com/practice/git/page/LoginPageObject.java:18]: Without wait for something?

File: src/test/java/com/practice/git/model/BaseModel.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.practice.git.model;
+
+/**
+ * @author anatoli.anukevich on 18:57 25-Mar-21
+ */
+public class BaseModel {
```
[the remark this answers is not in the document]
Comment (YummyMirror) [reply src/test/java/com/practice/git/model/BaseModel.java:6]: It's just BaseModel for generic purposes

File: src/test/java/com/practice/git/annotation/Source.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * @author anatoli.anukevich on 18:57 25-Mar-21
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface Source {
+    String value() default "";
+
+    Class<? extends BaseModel> model() default BaseModel.class;
```
[the remark this answers is not in the document]
Comment (YummyMirror) [reply src/test/java/com/practice/git/annotation/Source.java:16]: Yes, only for DP purposes

File: src/test/java/com/practice/git/model/User.java [hunk 5]
```diff
@@ -6,6 +6,7 @@
 public class User extends BaseModel {
     private String username;
     private String password;
+    private boolean remember;
```
[the remark this answers is not in the document]
Comment (YummyMirror) [reply src/test/java/com/practice/git/model/User.java:9]: Yes, for Cookies

File: src/test/java/com/practice/git/page/LoginPageObject.java (hunk 2 above)
Comment (YummyMirror) [reply src/test/java/com/practice/git/page/LoginPageObject.java:18]: Wait will be using in certain method before doing something

File: src/test/java/com/practice/git/page/Navigator.java (hunk 1 above)
Comment (YummyMirror) [reply src/test/java/com/practice/git/page/Navigator.java]: Add PropertyUtils and removed hardcode

**Pull request merged**